runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingPortablePipelineTranslator.java

```diff
@@ -187,11 +217,152 @@ public void flatMap(T t, Collector<T> collector) throws Exception {
+    RunnerApi.WindowingStrategy windowingStrategyProto =
+            pipeline.getComponents().getWindowingStrategiesOrThrow(
+                    pipeline.getComponents().getPcollectionsOrThrow(
+                            inputPCollectionId).getWindowingStrategyId());
+
+    //KvCoder<K, InputT> inputKvCoder = (KvCoder<K, InputT>) input.getCoder();
+    RunnerApi.Coder inputCoderProto =
+            pipeline.getComponents().getCodersOrThrow(
+                    pipeline.getComponents().getPcollectionsOrThrow(
+                            inputPCollectionId).getCoderId());
+
+    //DataStream<WindowedValue<KV<K, InputT>>> inputDataStream = context.getInputDataStream(input);
+    DataStream<WindowedValue<KV<K, V>>> inputDataStream =
+            context.getDataStreamOrThrow(inputPCollectionId);
+
+    RunnerApi.Coder outputCoderProto =
+            pipeline.getComponents().getCodersOrThrow(
+                    pipeline.getComponents().getPcollectionsOrThrow(
+                            outputPCollectionId).getCoderId());
+
+    RehydratedComponents rehydratedComponents =
+            RehydratedComponents.forComponents(pipeline.getComponents());
+    KvCoder<K, V> inputCoder;
+    try {
+      inputCoder = (KvCoder<K, V>) CoderTranslation.fromProto(
+              inputCoderProto,
+              rehydratedComponents);
+    } catch (IOException e) {
+      throw new IllegalStateException(
+              String.format("Unable to hydrate GroupByKey input coder %s.",
+                      inputCoderProto),
+              e);
+    }
+
+    WindowingStrategy<Object, BoundedWindow> windowingStrategy;
+    try {
+      windowingStrategy =
+              (WindowingStrategy<Object, BoundedWindow>) WindowingStrategyTranslation.fromProto(
+                      windowingStrategyProto,
+                      rehydratedComponents);
+    } catch (InvalidProtocolBufferException e) {
+      throw new IllegalStateException(
+              String.format("Unable to hydrate GroupByKey windowing strategy %s.",
+                      windowingStrategyProto),
+              e);
+    }
+
+
+    SingletonKeyedWorkItemCoder<K, V> workItemCoder = SingletonKeyedWorkItemCoder.of(
+            inputCoder.getKeyCoder(),
+            inputCoder.getValueCoder(),
+            windowingStrategy.getWindowFn().windowCoder());
+
+    WindowedValue.
+            FullWindowedValueCoder<SingletonKeyedWorkItem<K, V>> windowedWorkItemCoder =
+            WindowedValue.getFullCoder(
+                    workItemCoder,
+                    windowingStrategy.getWindowFn().windowCoder());
+
+    CoderTypeInformation<WindowedValue<SingletonKeyedWorkItem<K, V>>> workItemTypeInfo =
+            new CoderTypeInformation<>(windowedWorkItemCoder);
+
+    DataStream<WindowedValue<SingletonKeyedWorkItem<K, V>>> workItemStream =
+            inputDataStream
+                    .flatMap(new FlinkStreamingTransformTranslators.ToKeyedWorkItem<>())
```

Comment:
This is strange. All of the batch transform Flink functions are separated into their own classes. It might be worth doing that here.